src/assetmanagement/util.rs

```diff
@@ -57,3 +58,37 @@ pub fn insert_into_asset_storages(entity: Entity, asset_name: &str, storages: &m
     mat_storage.insert(entity, material).unwrap();
     mesh_handles.insert(entity, mesh).unwrap();
 }
+
+use systems::HoverHandler;
+
+pub fn add_hover_handler<'a>(
+    entity: Entity,
+    asset_name: &str,
+    bounding_box: Primitive3<f32>,
+    (loader, tex_manager, tex_storage, default_mat, hover_storage): &mut (
+        ReadExpect<'a, Loader>,
+        Write<'a, AssetManager<Texture>>,
+        Write<'a, AssetStorage<Texture>>,
+        ReadExpect<'a, MaterialDefaults>,
+        WriteStorage<'a, HoverHandler>,
+    ),
+) {
+    let hover_mat = {
+        let handle = tex_manager.get_asset_handle_or_load(
+            &[asset_name, "_hover"].join(""),
+            PngFormat,
+            TextureMetadata::srgb(),
+            tex_storage,
+            &loader,
+        );
+        Material {
+            albedo: handle,
+            ..default_mat.0.clone()
+        }
```

Comment:
It does, and there is no other possibility. Amethyst requires Ownership of the Material. Every Material is associated with one Entity. I can ask, but afaik, there is no other way.

Since a [Material](https://www.amethyst.rs/doc/latest/doc/amethyst_renderer/struct.Material.html) contains only Handles, not the Textures, I see no problem with that